modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/IgnitePlanner.java

```diff
@@ -268,10 +272,12 @@ public ValidationResult validateAndGetTypeMetadata(SqlNode sqlNode) {
                         if (cnt < selectItemsNoStar.size()) {
                             SqlNode noStarItem = selectItemsNoStar.get(cnt);
 
+                            // The validator can transform SqlIdentifier to AS call. We should check whether
+                            // AS call is a real one and take the second operand from original one.
                             if (noStarItem instanceof SqlBasicCall
-                                    && ((SqlBasicCall)node).operandCount() == 2 && node.getKind() == SqlKind.AS
-                                    && ((SqlBasicCall)node).operand(0) instanceof SqlIdentifier
-                                    && ((SqlBasicCall)node).operand(1) instanceof SqlIdentifier) {
+                                    && isAsCall((SqlBasicCall)noStarItem)
+                                    && node instanceof SqlBasicCall
```

Comment:
It's already checked earlier (we are in `if (node instanceof SqlBasicCall)` branch). Also, this check can be moved to `isAsCall`.